test/tests/serial/mail_hostname.bats

```diff
@@ -1,213 +1,260 @@
+  _should_have_expected_hostname 'mail'
+
+  _should_be_configured_to_domainname 'example.test'
+  _should_be_configured_to_fqdn 'mail.example.test'
+
+  # Likewise `--hostname` value will always match the third parameter:
+  _should_have_correct_mail_headers 'mail.example.test' 'example.test' 'mail'
 }
 
-@test "checking configuration: hostname/domainname override: check headers of received mail" {
-  run docker exec mail_override_hostname /bin/sh -c "ls -A /var/mail/localhost.localdomain/user1/new | wc -l | grep 1"
-  assert_success
-
-  run docker exec mail_override_hostname /bin/sh -c "cat /var/mail/localhost.localdomain/user1/new/* | grep mail.my-domain.com"
+# This test is purely for testing the ENV `SRS_DOMAINNAME` (not relevant to these tests?)
+@test "should give priority to ENV in postsrsd config (ENV SRS_DOMAINNAME)" {
+  export CONTAINER_NAME="${CONTAINER5_NAME}"
+
+  local CUSTOM_SETUP_ARGUMENTS=(
+    --hostname 'mail'
+    --domainname 'example.test'
+    --env ENABLE_SRS=1
+    --env SRS_DOMAINNAME='srs.example.test'
+    --env PERMIT_DOCKER='container'
+    --ulimit "nofile=$(ulimit -Sn):$(ulimit -Hn)"
+  )
+  _init_with_defaults
+  _common_container_setup 'CUSTOM_SETUP_ARGUMENTS'
+
+  # PostSRSd should be configured correctly:
+  _run_in_container_bash "grep '^SRS_DOMAIN=' /etc/default/postsrsd"
+  assert_output "SRS_DOMAIN=srs.example.test"
   assert_success
-
-  # test whether the container hostname is not found in received mail
-  run docker exec mail_override_hostname /bin/sh -c "cat /var/mail/localhost.localdomain/user1/new/* | grep unknown.domain.tld"
-  assert_failure
 }
 
-@test "checking SRS: OVERRIDE_HOSTNAME is handled correctly" {
-  run docker exec mail_override_hostname grep "SRS_DOMAIN=my-domain.com" /etc/default/postsrsd
-  assert_success
-}
+function _should_have_expected_hostname() {
+  local EXPECTED_FQDN=${1}
 
-@test "checking dovecot: postmaster address" {
-  run docker exec mail_override_hostname /bin/sh -c "grep 'postmaster_address = postmaster@my-domain.com' /etc/dovecot/conf.d/15-lda.conf"
+  _run_in_container_bash "hostname"
+  assert_output "${EXPECTED_FQDN}"
   assert_success
-}
 
-#
-# non-subdomain tests
-#
-
-@test "checking configuration: non-subdomain: check container hostname is applied correctly" {
-  run docker exec mail_non_subdomain_hostname /bin/bash -c "hostname | grep domain.com"
+  _run_in_container_bash "grep -E '[[:space:]]+${EXPECTED_FQDN}' /etc/hosts"
   assert_success
 }
 
-@test "checking configuration: non-subdomain: check overriden hostname is applied to all configs" {
-  run docker exec mail_non_subdomain_hostname /bin/bash -c "cat /etc/mailname | grep domain.com"
-  assert_success
+function _should_be_configured_to_domainname() {
+  local EXPECTED_DOMAIN=${1}
 
-  run docker exec mail_non_subdomain_hostname /bin/bash -c "postconf -n | grep mydomain | grep domain.com"
+  # setup-stack.sh:_setup_mailname
+  _run_in_container_bash "cat /etc/mailname"
```

Comment:
Could this also be:

```suggestion
  _run_in_container cat /etc/mailname
```

? I am not 100% (but 99%) sure, but my approach should work too and it should also be preferred. Only because we have the `_bash` command does not mean we need to execute programs that are perfectly in `$PATH` with it 😄